validation-test/compiler_crashers_2_fixed/0042-rdar21775089.swift

```diff
@@ -7,7 +7,7 @@ protocol MySequenceType {}
 protocol MyIndexableType {}
 
 protocol MyCollectionType : MySequenceType, MyIndexableType {
-  typealias SubSequence = MySlice<Self>
+  associatedtype SubSequence = MySlice<Self>
```

Comment:
The test case itself was ill-formed because of clashing `typealias` definition, but we didn't catch it before because the GSB was ignoring the inconsistency. I believe the intent here was to use `associatedtype` with defaults, because the test itself was intended to be well-formed and go all the way to IRGen. This change makes the test case compile again.